taroscript/send.go

```diff
@@ -152,6 +153,107 @@ func PrepareOutputAssets(input *taropsbt.VInput,
 	return nil
 }
 
+// SignVirtualTransaction updates the new asset (the root asset located at the
+// change output in case of a non-interactive or partial amount send or the
+// full asset in case of an interactive full amount send) by creating a
+// signature over the asset transfer, verifying the transfer with the Taro VM,
+// and attaching that signature to the new Asset.
+func SignVirtualTransaction(input *taropsbt.VInput, outputs []*taropsbt.VOutput,
+	signer Signer, validator TxValidator) error {
+
+	prevAssets := commitment.InputSet{
+		input.PrevID: input.Asset,
+	}
+	newAsset := outputs[1].Asset
+	if input.IsSplit {
+		newAsset = outputs[0].Asset
+	}
+
+	// Create a Taro virtual transaction representing the asset transfer.
+	virtualTx, _, err := VirtualTx(newAsset, prevAssets)
+	if err != nil {
+		return err
+	}
+
+	// For each input asset leaf, we need to produce a witness. Update the
+	// input of the virtual TX, generate a witness, and attach it to the
+	// copy of the new Asset.
+	//
+	// TODO(guggero): I think this is wrong... We shouldn't look at
+	// PrevWitnesses of the single asset we spend but instead have multiple
+	// inputs if we want to spend multiple coins of the same asset?
+	prevWitnessCount := len(newAsset.PrevWitnesses)
```

Comment:
Yeah so we need to sign the new asset (which also anchors the split), along with the other leaves that occupied that same prev anchor output. 

Here we're instead, re-signing all the old inputs for a merge (iiuc) which isn't correct. 

In addition to just the `Vinput`, we also need the `TaroCommitment` for that, as for now we need to re-sign all the other inputs. FWIW, we only support spending a single input for now, so things are restricted to just what was in that sole tree.

Reply:
Ah ok this was lifted from the old code. It's actually partially untested, since we only support a single input. This would be for the "merge" case, where we need to sign more than one taro input. 

Once we lift the merge restriction (ends up needing a diff proof file creation), then this is equivalent to signing multiple Bitcoin inputs when you need to consolidate change to make a spend.

Reply:
Yeah, I'll update the TODO to mention the merge case.